addons/survey/static/src/js/survey_choice_image_zoomer.js

```diff
@@ -0,0 +1,104 @@
+odoo.define('survey.choice_image_zoomer', function (require) {
+    'use strict';
+
+var publicWidget = require('web.public.widget');
+
+publicWidget.Widget.surveyChoiceImageZoomer = publicWidget.Widget.extend({
+    template: 'survey.survey_choice_image_zoomer',
+    xmlDependencies: ['/survey/static/src/xml/survey_choice_image_zoomer_templates.xml'],
+    events: {
+        'DOMMouseScroll .o_survey_img_zoomer_image': '_onImgScroll', // Firefox
+        'mousewheel .o_survey_img_zoomer_image': '_onImgScroll', // Chrome, Safari, IE
+        'click': '_onZoomerClick', // Allow user to close by clicking anywhere (mobile...)
+        'click .o_survey_img_zoomer_in_btn': '_onZoomIn',
+        'click .o_survey_img_zoomer_out_btn': '_onZoomOut',
+    },
+    /**
+     * @override
+     */
+    init(params) {
+        this.zoomImageScale = 1;
+        this.zoomStep = 0.1;
+        // The image is needed to render the template survey_choice_image_zoomer.
+        this.sourceImage = params.sourceImage;
+        return this._super(... arguments);
+    },
+    /**
+     * Open a transparent modal displaying the survey choice image.
+     * @override
+     */
+    async start() {
+        let self = this;
+        await this._super(...arguments).then(() => {
+            self.$el.modal({
+                keyboard: true,
+                backdrop: false,
+            });
+            // When hiding the modal with ESC, we want it deleted.
+            self.$el.on('hidden.bs.modal', () => self.destroy());
+            self.$el.modal('show');
+        });
+    },
+    /**
+     * Zoom in/out image on scrolling
+     *
+     * @private
+     * @param {MouseEvent} e
+     */
+    _onImgScroll(e) {
+        e.preventDefault();
+        if (e.originalEvent.wheelDelta > 0 || e.originalEvent.detail < 0) {
+            this._addZoomSteps(1);
+        } else {
+            this._addZoomSteps(-1);
+        }
+    },
+    /**
+     * Destroying the modal without using 'hide' would leave a modal-open in the view.
+     * @private
+     * @param {Event} e
+     */
+     _onZoomerClick(e) {
+        e.preventDefault();
+        this.$el.modal('hide');
+    },
+    /**
+     * @private
+     * @param {Event} e
+     */
+    _onZoomIn(e) {
+        e.stopPropagation();
+        this._addZoomSteps(1);
+    },
+    /**
+     * @private
+     * @param {Event} e
+     */
+    _onZoomOut(e) {
+        e.stopPropagation();
+        this._addZoomSteps(-1);
+    },
+    /**
+     * Zoom in / out the image by changing the scale by the given number of steps.
+     *
+     * @private
+     * @param {integer} zoomStepNumber - Number of times zoomStep is applied to the scale of
+     * the image. It can be negative, in order to zoom out.
+     */
+     _addZoomSteps(zoomStepNumber) {
+        const $image = this.$('.o_survey_img_zoomer_image');
+        const $zoomer = this.$('.o_survey_img_zoomer_body');
```

Comment:
Could try to avoid jQuery if possible by using native JS ("querySelector").

Reply:
I've had a look, we can but as there is calculation going on, it seems preferable to use jquery width and height functions that return a unit-less pixel measure. (The alternative would be .css('width') and .css('height'), returning a measure with units, like 450px)

Side note: I'm guessing this is a performance issue? Native is always faster? :)

Reply:
Not sure about calculation, looks like there is some "clientWidth" and "offsetWidth" native properties used in the current code (and it's height counterpart).
Would have to be tested though, I never used those myself.

About the side-note: I know one of the end-goal is to remove the dependency to jQuery in Odoo.
Although it's a long-term kind of thing, it's always better to avoid introducing new code using it if possible.

Cheers!

Reply:
Those seem like nice properties indeed. This can be also useful when dealing with borders, overflow, margins... Pretty cool. In this case it works as well.